core/darwin/src/Converters.kt

```diff
@@ -82,3 +91,12 @@ public fun LocalDateTime.toNSDateComponents(): NSDateComponents {
     components.nanosecond = nanosecond.convert()
     return components
 }
+
+/**
+ * Converts the given [LocalDateTime] to [NSDate].
+ */
+public fun LocalDateTime.toNSDate(): NSDate? {
+    val components = toNSDateComponents()
+    components.calendar = NSCalendar.currentCalendar
```

Comment:
> That said, it seems like, even in the Foundation library, the calendar is always explicitly passed, right?

Yes.

> Maybe a good way to fix this would be to allow passing an NSCalendar as a parameter.

What about making this a parameter with a default parameter?
```
public fun LocalDateTime.toNSDate(calendar: NSCalendar = NSCalendar.currentCalendar): NSDate? {
```

I'm not sure if this is supported in kmm or not.

Reply:
> This feels against the spirit of our library. We strictly separate Instant and LocalDateTime and require explicitness everywhere: passing TimeZone objects, not being able to do arithmetic on plain LocalDateTime objects, etc. We purposely don't have any hidden state that affects the results of any operations. Here, the result will depend on the current calendar.

I understand this, but I'm trying to improve the usability of this library in iOS.
I think that I would like to see improved in kmm in general is the usability from iOS.
Using kotlin kmm is so much easier from Android than from iOS, that this makes it even harder to convince the iOS devs to use this.
But I think we are in a good path.

One thing that I couldn't figure out by myself is how to access this class `Converters.kt` from iOS.
I needed to a public api in my shared library that just calls this convert to expose it to Swift.
Am I doing something wrong?

Reply:
The default parameter solution also doesn't feel idiomatic: for example, we don't default to `TimeZone.currentSystemDefault`. The reason is that it's very easy to accidentally omit the parameter if it's optional and in some cases acquire results that are slightly wrong.

> I understand this, but I would like to improve the usability of this library in iOS.

We would like to as well! The question is, would these methods improve the usability or hinder it? If these methods are easy to misuse, they could be more trouble than they're worth, don't you agree?

> One thing that I couldn't figure out by myself is how to access this class Converters.kt from iOS.

Maybe https://github.com/Kotlin/kotlinx-datetime/issues/78#issuecomment-740573900 is relevant (though not exactly the same)? In general, definitions from libraries won't be exported by default unless they are used in the Kotlin code.

Reply:
> We would like to as well! The question is, would these methods improve the usability or hinder it? If these methods are easy to misuse, they could be more trouble than they're worth, don't you agree?

I see your point.
It's hard to draw the line on a place to make everyone happy.
I will make the changes your suggested.

> Maybe https://github.com/Kotlin/kotlinx-datetime/issues/78#issuecomment-740573900 is relevant (though not exactly the same)? In general, definitions from libraries won't be exported by default unless they are used in the Kotlin code.

Thanks, that will help 👍

Reply:
Can you please take a look at the last changes to see if they address all of your comments?
Thanks